akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -635,9 +635,9 @@ public String apply(String aggr, String in) throws Exception {
   public void mustBeAbleToUseBatchWeighted() throws Exception {
     final TestKit probe = new TestKit(system);
     final List<String> input = Arrays.asList("A", "B", "C");
-    final Flow<String, String, NotUsed> flow = Flow.of(String.class).batchWeighted(3L, new Function<String, Object>() {
+    final Flow<String, String, NotUsed> flow = Flow.of(String.class).batchWeighted(3L, new Function<String, java.lang.Long>() {
       @Override
-      public Object apply(String s) throws Exception {
```

Comment:
Yup, seems like there was some misunderstanding in the way how it should work.
But, now the type-safety-net has been restored :)